### ISSUE TITLE
Better Error Handling for Bad PyPIs

### ISSUE THREAD

**Issue by CF-FHB-X** [astral-sh/uv#7257]
Hi there!

I'm trying to move over from `Poetry` to `uv`, but I'm running into an issue with one of the internal PyPIs that our Nexus is proxying. When I run `uv add package_name`, I get the following error:

```
error: Received some unexpected HTML from https://url.to.nexus/repository/pypi/simple/package_name/
  Caused by: Unexpected fragment (expected '#sha256=...' or similar) on URL: main
```

For some reason that downstream repo has a bunch of "fluff" for each package's endpoint (i.e., instead of just listing all the packages with their hashes, it also has links like "About", "Documentation", etc.). I suspect it's because that PyPI is a custom deployment (caveat: I have zero input on our Nexus, and even less than zero input on that downstream custom PyPI). If I don't include any packages that reside on that custom downstream repo, `uv` works _great_, but the moment I need anything on that custom repo, bam. Errors.

Is there some setting for `uv` where it can ignore situations like this? Something along the lines of a `--ignore-fragments` or something more appropriate?

I'm running `uv` v0.4.7 and running on Windows 11. I've tested this on Python 3.10 through 3.12.

Thanks!

**Comment by CF-FHB-X** [comment]
Just curious if there's any extra info on whether this can be added to a future version or not? If there's something I can help with, more than happy to!

**Comment by mulliken** [comment]
I think I am running into the same issue and would be interested in any updates.

**Comment by zanieb** [comment]
@mulliken can you share the error? Without providing more details about your situation, we can't use your comment as further motivation for adding this feature.

@CF-FHB-X Sorry for the lack of reply. It seems generally hard to support non-compliant indexes. I think we're only likely to take action here if see more use-cases and have a better understanding of the patterns present in these indexes. Does pip work with this index?

**Comment by CF-FHB-X** [comment]
@zanieb Yes, `pip` and `Poetry` both work with that particular repo without any issues. My feeling is that `uv` is seeing those non-package HTML links and isn't able to parse them correctly, and hence the `unexpected HTML` error. Maybe I'm oversimplifying it, but would it just be an issue of adding in some error handling for that particular part where a non-standard HTML link that can't be parsed is just ignored (or instead throws a warning instead of an error)? 

I'm sure there's more going on in the backend that I'm not aware of, but I also know how those Nexus repos have caused issues in the past with indexes that are either out of date or have those extra HTML links in there. Both `pip` and `Poetry` seem to be able to handle them (although `Poetry` doesn't like out of date indexes where files are missing).

Hope that helps!

**Comment by CF-FHB-X** [comment]
Awesome - thanks for that fix! I'll keep you posted when that version is released.

**Comment by CF-FHB-X** [comment]
Sorry for the delay in replying (got sidetracked with a couple of other things), but everything is working like a charm now! 

Thanks again!
